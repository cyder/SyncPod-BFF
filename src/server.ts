import { ApolloServer } from 'apollo-server-express';
import express from 'express';

import resolvers from './resolvers';
import typeDefs from './typeDefs';
import SyncpodV1Api from './api/syncpodV1Api';

const CLIENT_HOST = new RegExp(process.env.CLIENT_HOST || '^http://localhost(:[0-9]+)?$');

const app = express();
const server = new ApolloServer({
  typeDefs,
  resolvers,
  dataSources: () => ({
    syncpodV1API: new SyncpodV1Api(),
  }),
});

server.applyMiddleware({ app, cors: { credentials: true, origin: CLIENT_HOST } });

export default app;
export { server };
