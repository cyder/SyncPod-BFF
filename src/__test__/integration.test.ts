import { createTestClient } from 'apollo-server-testing';
import { gql } from 'apollo-server-express';

import { server } from '../server';

const PING = gql`
  query {
    ping
  }
`;

const LOGIN = gql`
  mutation login($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      id
      icon
      email
      name
      accessToken
      createdAt
      updatedAt
      selfIntroduction
    }
  }
`;

describe('Queries', () => {
  it('returns pong ', async () => {
    const { query } = createTestClient(server);
    const res = await query({ query: PING });
    expect(res).toMatchSnapshot();
  });
});

describe('Mutations', () => {
  it('returns user', async () => {
    const { mutate } = createTestClient(server);
    const res = await mutate({
      query: undefined,
      mutation: LOGIN,
      variables: { email: 'user@example.com', password: 'password' },
    } as any);
    expect(res).toMatchSnapshot();
  });
});
