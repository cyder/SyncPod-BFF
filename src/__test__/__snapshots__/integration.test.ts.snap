// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutations returns user 1`] = `
Object {
  "data": Object {
    "login": Object {
      "accessToken": "69:q1aXhT-onz4sWejxcV1s",
      "createdAt": "2018-03-31T07:35:05.000Z",
      "email": "user@example.com",
      "icon": null,
      "id": 69,
      "name": "MyString",
      "selfIntroduction": null,
      "updatedAt": "2018-03-31T07:35:05.000Z",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries returns pong  1`] = `
Object {
  "data": Object {
    "ping": "pong",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
